state/postgresql/postgresdbaccess.go

```diff
@@ -246,32 +265,37 @@ func (p *PostgresDBAccess) doSet(parentCtx context.Context, db dbquerier, req *s
 	} else {
 		queryExpiredate = "NULL"
 	}
-	result, err = db.ExecContext(parentCtx, fmt.Sprintf(query, p.metadata.TableName, queryExpiredate), params...)
 
+	result, err := db.Exec(parentCtx, fmt.Sprintf(query, p.metadata.TableName, queryExpiredate), params...)
 	if err != nil {
 		if req.ETag != nil && *req.ETag != "" {
 			return state.NewETagError(state.ETagMismatch, err)
 		}
 		return err
 	}
 
-	rows, err := result.RowsAffected()
-	if err != nil {
-		return err
-	}
-	if rows != 1 {
+	if result.RowsAffected() != 1 {
 		return errors.New("no item was updated")
 	}
 
 	return nil
 }
 
 func (p *PostgresDBAccess) BulkSet(parentCtx context.Context, req []state.SetRequest) error {
-	tx, err := p.db.BeginTx(parentCtx, nil)
+	ctx, cancel := context.WithTimeout(parentCtx, 30*time.Second)
```

Comment:
can we have all timeouts be defined as constants at least? no magic numbers please.

Reply:
the same question here, should this be a configurable param?